nautobot/extras/tests/test_views.py

```diff
@@ -909,3 +923,361 @@ def test_create_object_with_constrained_permission(self):
         for cf in CustomField.objects.all():
             cf.delete()
         super().test_create_object_with_constrained_permission()
+
+
+@override_settings(JOBS_ROOT=THIS_DIRECTORY)
+class TestJobMixin(SimpleTestCase):
+    class TestJob(Job):
+        pass
+
+    def get_test_job_class(self, class_path):
+        if class_path.startswith("local/test_view"):
+            return self.TestJob
+        raise Http404
+
+    def setUp(self):
+        super().setUp()
+
+        # Monkey-patch the viewsets' _get_job methods to return our test class above
+        JobView._get_job = self.get_test_job_class
+        ScheduledJobView._get_job = self.get_test_job_class
```

Comment:
Not that this is a huge issue, but is this necessary as part of every test method or could this be sufficient at the test case (class) level using `setUpClass()` instead? Just thinking about test efficiency.